truth_set/java/structure/organisation/complex-code/ECommerce.java

```diff
@@ -0,0 +1,224 @@
+                continue;
+            }
+
+            double itemTotal = 0;
+            Double productPrice = getProductPrice(item.getProductId());
+            if (productPrice == null) {
+                items.put("item_" + i, "Price Not Available");
+                continue;
+            }
+
+            itemTotal += productPrice * item.getQuantity();
+
+            if (item.getOptions() != null) {
+                for (Option option : item.getOptions()) {
+                    if (option.getOptionId() == null || option.getValue() == null) {
+                        continue;
+                    }
+                    Double optionPrice = getOptionPrice(option.getOptionId(), option.getValue());
+                    if (optionPrice != null) {
+                        itemTotal += optionPrice;
+                    }
+                }
+            }
+
+            if (item.getDiscounts() != null) {
+                for (Discount discount : item.getDiscounts()) {
+                    if (discount.getDiscountId() == null || discount.getAmount() <= 0) {
+                        continue;
+                    }
+                    itemTotal -= discount.getAmount();
+                }
+            }
+
+            Map<String, Object> itemDetails = new HashMap<>();
+            itemDetails.put("productId", item.getProductId());
+            itemDetails.put("quantity", item.getQuantity());
+            itemDetails.put("total", Math.max(itemTotal, 0));
+            items.put("item_" + i, itemDetails);
+
+            if (order.getCoupons() != null) {
+                for (Coupon coupon : order.getCoupons()) {
+                    if (coupon.getCode() == null || !coupon.isValid() || 
+                        new Date(coupon.getExpiryDate()).before(new Date())) {
+                        continue;
+                    }
+                    ((Map<String, Object>)items.get("item_" + i)).put("coupon", coupon.getCode());
+                }
+            }
+        }
+
+        if (order.getShipping() != null && order.getShipping().getMethod() != null) {
+            if (order.getShipping().getCost() <= 0) {
+                finalOrder.put("shipping", "Method: " + order.getShipping().getMethod() + 
+                             ", Cost: Calculating");
+            } else {
+                finalOrder.put("shipping", "Method: " + order.getShipping().getMethod() + 
+                             ", Cost: $" + order.getShipping().getCost());
+            }
+        } else {
+            finalOrder.put("shipping", "Standard Shipping");
+        }
+
+        if (((Map<String, Object>)finalOrder.get("items")).isEmpty()) {
+            return null;
+        }
+
+        return finalOrder;
+    }
+
+    private Double getProductPrice(String productId) {
+        Map<String, Double> priceList = new HashMap<>();
+        priceList.put("P001", 29.99);
+        priceList.put("P002", 49.99);
+        priceList.put("P003", 19.99);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider moving product prices to config</b></div>
<div id="fix">

Consider moving the product price list to a configuration class or database. Hardcoding prices in the method makes updates difficult and error-prone.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
        Map<String, Double> priceList = PriceConfiguration.getProductPrices();
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2530735434>#cc25d3</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged